app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -498,6 +506,15 @@ export default class ComposerController extends Controller {
     });
   }
 
+  @action
+  setCategory(categoryId) {
+    this.model.categoryId = categoryId;
```

Comment:
Do we need to do this? Isn't the category id of the model already the same?

Reply:
This is for when we update the category in the picker. I think I should change the name to something more fitting like `updateCategory(categoryId)` instead.

I changed it to an `@action` in the JS file instead of the inline action:
```suggestion
    @onChange={{action(mut this.composer.model.categoryId)}}
    @onChange={{this.composer.updateCategory}}
```

because I wanted to also trigger the form templates to reload with the updated id

```js
    if (this.formTemplateIds?.length) {
      this.appEvents.trigger("composer:load-templates", this.formTemplateIds);
    }
 ```